docs/introduction-to-smart-contracts.rst

```diff
@@ -504,10 +511,8 @@ Delegatecall / Callcode and Libraries
 
 There exists a special variant of a message call, named **delegatecall**
 which is identical to a message call apart from the fact that
-the code at the target address is executed in the context of the calling
-contract. In practice this means ``msg.sender`` and ``msg.value`` retain
-the values that were passed to the delegator, while executing code that
-lives in the delegated contract.
+the code at the target address is executed in the context (i.e. at the address) of the calling
+contract and ``msg.sender`` and ``msg.value`` do not change thair values.
```

Comment:
Since this is an intro, might be worth adding a few words to clarify the consequences:
```suggestion
contract and ``msg.sender`` and ``msg.value`` retain the values they have in that context,
effectively letting the callee impersonate the calling contract and freely modify its state.
```

Reply:
@ekpyron What do you think about this one? Once we resolve this, I could fix the two typos, merge and we'd be one PR lighter :)

Reply:
Hm... "context" is a bit underdefined for that :-). Earlier we say by "context" we mean "at the address of the calling
contract"... but `msg.sender` and `msg.value` don't have values "at that address"... they retain the values they had at the call site...

And in general the sentence is getting too long... I'd rather keep the sentence as in the PR now - and we can think about adding a new sentence about "effectively allowing to impersonate"... on the other hand the next paragraph also kind of says that...

Reply:
> on the other hand the next paragraph also kind of says that...

A, right. Didn't look too far ahead :) OK, let's drop this suggestion in that case.